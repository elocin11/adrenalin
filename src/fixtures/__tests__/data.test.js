import data from "./../data";

// match the data fixture to the generated snapshot
// should have 4 items
test("data fixtures is correct", () => {
  expect(data).toMatchSnapshot();
  expect(data).toHaveLength(4);
});

// check the property of each items
data.map((item) =>
  test(`${item} should have properties (id, title, title_long, tag, thumb, image, slug, questions)`, () => {
    expect(item).toHaveProperty("id");
    expect(item).toHaveProperty("title");
    expect(item).toHaveProperty("title_long");
    expect(item).toHaveProperty("tag");
    expect(item).toHaveProperty("thumb");
    expect(item).toHaveProperty("image");
    expect(item).toHaveProperty("slug");
    expect(item).toHaveProperty("questions");
  })
);
