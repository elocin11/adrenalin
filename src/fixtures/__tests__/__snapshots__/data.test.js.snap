// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`data fixtures is correct 1`] = `
Array [
  Object {
    "id": 1001,
    "image": "1001.jpg",
    "questions": Array [
      "Donec posuere porta blandit. Suspendisse fringilla lacus ac imperdiet pretium. Nullam ultricies velit ut nisi finibus eleifend.",
      "Donec tempus dolor in lobortis venenatis. Pellentesque sit amet dictum ligula, eu facilisis nisl. Nulla facilisi.",
      "Vestibulum bibendum dolor non lacus consectetur bibendum. Nulla vel ullamcorper tortor. Curabitur fermentum faucibus sem, eu laoreet sem congue nec",
      "Vestibulum nec enim erat. Cras gravida blandit libero, malesuada vestibulum arcu pulvinar eget. Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
      "Cras vel nunc viverra, ornare purus a, tempor enim. Mauris dictum, est non consectetur elementum, velit purus varius lectus, vitae laoreet mauris velit id nibh. In at justo nec erat blandit tincidunt.",
    ],
    "slug": "lorem-ipseum",
    "tag": "Toga Group",
    "thumb": "1001_thumb.jpg",
    "title": "Lorem ipsum dolor sit amet",
    "title_long": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed ut libero id auctor.",
  },
  Object {
    "id": 1002,
    "image": "1002.jpg",
    "questions": Array [
      "Curabitur id enim ornare, porta est id, feugiat mauris.",
      "Donec tempus dolor in lobortis venenatis. Pellentesque sit amet dictum ligula, eu facilisis nisl. Nulla facilisi.",
      "Mauris tristique semper massa, at sagittis tortor vestibulum vel. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae",
    ],
    "slug": "aenean-vulputate",
    "tag": "Hungry Jacks",
    "thumb": "1002_thumb.jpg",
    "title": "Aenean vulputate augue sed",
    "title_long": "Aenean vulputate augue sed lorem ultricies, a placerat mi dictum",
  },
  Object {
    "id": 1003,
    "image": "1003.jpg",
    "questions": Array [
      "Pellentesque volutpat, eros at interdum sagittis, dolor lectus euismod nisl, sagittis laoreet risus urna aliquam nibh.",
      "Morbi sed orci erat. Sed in semper tortor.",
      "Phasellus volutpat, est ac rhoncus placerat, risus nulla consectetur dolor, facilisis eleifend sem nulla sed leo.",
    ],
    "slug": "nulla-non",
    "tag": "UNSW",
    "thumb": "1003_thumb.jpg",
    "title": "Nulla non ante ultrices",
    "title_long": "Nulla non ante ultrices, blandit velit non",
  },
  Object {
    "id": 1004,
    "image": "1004.jpg",
    "questions": Array [
      "Integer at interdum lectus. Cras rutrum quis purus eget iaculis.",
      "Praesent finibus est ante, nec porttitor diam efficitur ut. Suspendisse fermentum vulputate elementum",
      "Proin dictum velit elit, in venenatis metus lobortis a. Fusce nec nisl at leo vehicula maximus eget in ligula. Cras non sapien justo.",
      "In tempus lacus nisl, sed dignissim mi dapibus in. Phasellus sem sapien, sodales ut risus et, gravida congue libero.",
    ],
    "slug": "vivamus-vitae",
    "tag": "Osko",
    "thumb": "1004_thumb.jpg",
    "title": "Vivamus vitae elit viverra",
    "title_long": "Vivamus vitae elit viverra, imperdiet quam quis, fermentum justo",
  },
]
`;
